generators/cypress/templates/src/test/javascript/cypress/support/oauth2.ts.ejs

```diff
@@ -0,0 +1,104 @@
+
+/* eslint-disable @typescript-eslint/camelcase */
+/* eslint-disable @typescript-eslint/no-namespace */
+/* eslint-disable @typescript-eslint/no-use-before-define */
+// eslint-disable-next-line spaced-comment
+/// <reference types="cypress" />
+
+Cypress.Commands.add('getOauth2Data', () => {
+  cy.request({
+    method: 'GET',
+    url: '/oauth2/authorization/oidc',
+    followRedirect: false,
+  }).then(response => {
+    const url = new URL(response.headers['location']);
+    const clientId = url.searchParams.get('client_id');
+    const responseType = url.searchParams.get('response_type');
+    const redirectUri = url.searchParams.get('redirect_uri');
+    const nonce = url.searchParams.get("nonce");
+    const state = url.searchParams.get("state");
+    const scope = url.searchParams.get("scope");
+    const origin = url.origin;
+
+    const authenticationUrl = `${origin}${url.pathname}`;
+
+    let logoutUrl = "";
+    if (isOkta) {
+      logoutUrl = `${origin}${url.pathname.split('/',2).join('/')}/v1/logout`
+    } else {
+      logoutUrl = `${origin}${url.pathname.split('/', 4).join('/')}/protocol/openid-connect/logout`
+    }
+
+    const data = {
+      url,
+      clientId,
+      responseType,
+      scope,
+      redirectUri,
+      nonce,
+      state,
+      authenticationUrl,
+      logoutUrl,
+      isOkta
+    };  
+    cy.wrap(data).as('oauth2Data');
+  });
+});
+
+Cypress.Commands.add('oauthLogin', (oauth2Data: any, username: string, password: string) => {
+  if (oauth2Data.isOkta) {
+    cy.oktaLogin(username, password);
+  } else {
+    cy.keycloakLogin(oauth2Data, username, password);
```

Comment:
`keycloakLogin` Is missing.
If possible to use `authenticationUrl` instead of getting from form action would be great.